Contentful.Core/Models/Authoring.cs

```diff
@@ -0,0 +1,419 @@
+        /// </summary>
+        /// <param name="content">The content to evaluate.</param>
+        /// <returns>Returns true if the content is a heading, otherwise false.</returns>
+        public bool SupportsContent(IContent content)
+        {
+            return content is Heading;
+        }
+
+        /// <summary>
+        /// Renders the content to an html h-tag.
+        /// </summary>
+        /// <param name="content">The content to render.</param>
+        /// <returns>The p-tag as a string.</returns>
+        public string Render(IContent content)
+        {
+            var heading = content as Heading;
+            var sb = new StringBuilder();
+            sb.Append($"<h{heading.HeadingSize}>");
+
+            foreach (var subContent in heading.Content)
+            {
+                var renderer = _renderererCollection.GetRendererForContent(subContent);
+                sb.Append(renderer.Render(subContent));
+            }
+
+            sb.Append($"</h{heading.HeadingSize}>");
+            return sb.ToString();
+        }
+    }
+
+    /// <summary>
+    /// A renderer for a text node.
+    /// </summary>
+    public class TextRenderer : IContentRenderer
+    {
+        /// <summary>
+        /// The order of this renderer in the collection.
+        /// </summary>
+        public int Order { get; set; } = 100;
+
+        /// <summary>
+        /// Whether or not this renderer supports the provided content.
+        /// </summary>
+        /// <param name="content">The content to evaluate.</param>
+        /// <returns>Returns true if the content is a textual node, otherwise false.</returns>
+        public bool SupportsContent(IContent content)
+        {
+            return content is Text;
+        }
+
+        /// <summary>
+        /// Renders the content to a string.
+        /// </summary>
+        /// <param name="content">The content to render.</param>
+        /// <returns>The content as a string.</returns>
+        public string Render(IContent content)
+        {
+            var text = content as Text;
+            var sb = new StringBuilder();
+
+            if (text.Marks != null)
+            {
+                foreach (var mark in text.Marks)
+                {
+                    sb.Append($"<{MarkToHtmlTag(mark)}>");
+                }
+            }
+
+            sb.Append(text.Value);
+
+            if (text.Marks != null)
+            {
+                foreach (var mark in text.Marks)
+                {
+                    sb.Append($"</{MarkToHtmlTag(mark)}>");
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        private string MarkToHtmlTag(Mark mark)
+        {
+            switch (mark.Type)
+            {
+                case "bold":
+                    return "strong";
+                case "underline":
+                    return "u";
+                case "italic":
+                    return "em";
```

Comment:
I think you're missing `code`

Reply:
Probably a few more, right? I will add them once they're available in the web app.